### ISSUE TITLE
#[cfg] attrs in destructuring assignments are erroneously detected as syntax errors

### ISSUE THREAD

**Issue by hrydgard** [rust-analyzer/rust-analyzer#8608]
This compiles fine, but rust-analyzer detects syntax errors.

Summary of the issue in an image:
![image](https://user-images.githubusercontent.com/130929/115514346-4ad83c00-a284-11eb-946e-d7383962f5f3.png)

Copy-pasteable repro code:

```rust
fn main() {
    struct A {
        b: u32,
    }
    let a = A { b: 1 };
    let A {
        #[cfg(not(any()))]
        b
    } = a;
    println!("{}", b);
}
```

Quite an edge case, but there you go.

**Comment by hrydgard** [comment]
Thanks for the lightning fast fix!
